lib/HLSL/DxilContainerReflection.cpp

```diff
@@ -1128,6 +1128,13 @@ HRESULT CShaderReflectionType::Initialize(
         m_MemberTypes.push_back(fieldReflectionType);
         m_MemberNames.push_back(fieldAnnotation.GetFieldName().c_str());
 
+        // Skip structures fields with no real contents, otherwise we expand
+        // the size of this struct by 1 when we treat a zero column size as 1.
+        if (isa<StructType>(fieldType) &&
+            fieldReflectionType->m_Desc.Columns == 0) {
```

Comment:
Otherwise we think fields of empty struct type have Columns == 1, which messes up results.